phylanx/execution_tree/primitives/argmax.hpp

```diff
@@ -0,0 +1,40 @@
+//  Copyright (c) 2018 Parsa Amini
+//  Copyright (c) 2018 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(PHYLANX_PRIMITIVES_ARGMAX)
+#define PHYLANX_PRIMITIVES_ARGMAX
+
+#include <phylanx/config.hpp>
+#include <phylanx/execution_tree/primitives/base_primitive.hpp>
+#include <phylanx/execution_tree/primitives/primitive_component_base.hpp>
+
+#include <hpx/lcos/future.hpp>
+
+#include <string>
+#include <vector>
+
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    class argmax : public primitive_component_base
+    {
+    public:
+        static match_pattern_type const match_data;
+
+        argmax() = default;
+
+        argmax(std::vector<primitive_argument_type>&& operands,
```

Comment:
Could you add some documentation here, please?

Reply:
Please see the new commit.